articles/ajax-request-in-django-using-axios/index.md

```diff
@@ -0,0 +1,238 @@
+
+### Prerequisites
+
+1. [Python](https://www.python.org/downloads/) 3.7 or newer installed on your computer.
+
+2. Python package manager, [pip](https://pypi.org/project/pip/) installed on your computer.
+
+3. Knowledge of [Django](https://www.djangoproject.com/).
+
+3. Knowledge of [JavaScript](https://javascript.info/).
+
+### Goals.
+
+- Using Axios to submit a form.
+
+- Handling a post request in Django.
+
+### Project setup
+
+Ensure `virtualenv` is installed on your computer.
+
+```bash
+$ virtualenv --version
+
+virtualenv 20.2.2 from /home/username/.local/lib/python3.8/site-packages/virtualenv/__init__.py
+
+```
+
+If you get an error message command, run the command below to install `virtualenv` on your computer.
+
+```bash
+$ pip install virtualenv
+```
+
+Change to any directory of your choice on your machine or you can as well make a new directory.  Create a virtual environment for our project using `virtualenv`.
+
+ Run the command below to create and activate the virtual environment.
+
+```bash
+$ virtualenv venv
+
+$ source venv/bin/activate
+```
+
+I have created a starter template for you to follow along in this tutorial.
+Clone the starter template in your current directory using the command below.
+
+```bash 
+$ git clone https://github.com/atoyegbe/note-app.git
+ ```
+
+Run the command below to install the dependencies needed for this project to work.
+
+```bash 
+$ pip install -r requirements.txt
+```
+
+Let’s make sure nothing is broken in our application, change into the project directory and run the command below.
+
+```bash 
+$ python manage.py runserver
+```
+After running the above command you will see something like this in your browser.
+![form page](/engineering-education/ajax-request-in-django-using-axios/form_page.png)
+
+Let’s go into the **templates/home.html**.
+We will add **Axios** library to our project the by adding the CDN script tag below </div> closing tag.
+
+```html
+
+{% include ‘base.html’ %}
+
+{% block content %}
+
+<div class="container">
+
+     <h1> Notes </h1>
+
+ <!-- ..... -->
+ <!-- ..... -->
+
+</div>
+
+<!-- Adding axios script tag here-->
+
+<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.1/axios.min.js"></script>
+
+{% endblock %}
+
+```
+
+### Using Axios to submit a form.
+>Axios is a Javascript library used to make HTTP requests from node.js or XMLHttpRequests from the browser. It is a lightweight HTTP client based on the XMLHttpRequests service. It is similar to the Fetch API and is used to perform HTTP requests.
+
```

Comment:
Move this explanation somewhere before the code work. Create a sub-heading with the name "What is Axios" and put the explanation in it.  Also, provide relevant links for Fetch API and XMLHttpRequests.